ipyparallel/cluster/shellcmd.py

```diff
@@ -0,0 +1,860 @@
+                    continue
+                if p[0] == '"' and p[-1] == '"':
+                    start_cmd[idx] = p.strip('"')
+
+        self._log(f"start_cmd={start_cmd}  (use_break_way={self.use_break_way})")
+
+        if self.platform == Platform.Windows:
+            from subprocess import (
+                CREATE_BREAKAWAY_FROM_JOB,
+                CREATE_NEW_CONSOLE,
+                DEVNULL,
+                Popen,
+            )
+
+            flags = 0
+            if self.use_break_way != False:
+                flags |= CREATE_NEW_CONSOLE
+                flags |= CREATE_BREAKAWAY_FROM_JOB
+
+            pkwargs = {
+                'close_fds': True,  # close stdin/stdout/stderr on child
+                'creationflags': flags,
+            }
+            if output_file:
+                fo = open(output_file, "w")
+                pkwargs['stdout'] = fo
+                pkwargs['stderr'] = fo
+                pkwargs['stdin'] = DEVNULL
+
+            self._log(f"Popen(**pkwargs={pkwargs}")
+            p = Popen(start_cmd, **pkwargs)
+            self._log(f"pid={p.pid}")
+
+            print(f'__remote_pid={p.pid}__')
+            sys.stdout.flush()
+            if self.use_break_way == False:
+                self._log("before wait")
+                p.wait()
+                self._log("after wait")
+        else:
+            start_cmd = [self._linux_quote(x) for x in start_cmd]
+
+            if output_file:
+                nohup_start = f"nohup {' '.join(start_cmd)} >{output_file} 2>&1 </dev/null & echo __remote_pid=$!__"
+            else:
+                nohup_start = f"nohup {' '.join(start_cmd)} >/dev/null 2>&1 </dev/null & echo __remote_pid=$!__"
+            os.system(nohup_start)
+
+    def cmd_running(self, pid):
+        self._log(f"Check if pid {pid} is running")
+
+        if self.platform == Platform.Windows:
+            # taken from https://stackoverflow.com/questions/568271/how-to-check-if-there-exists-a-process-with-a-given-pid-in-python
+            import ctypes
+
+            PROCESS_QUERY_INFROMATION = (
+                0x1000  # if actually PROCESS_QUERY_LIMITED_INFORMATION
+            )
+            STILL_ACTIVE = 259
+            processHandle = ctypes.windll.kernel32.OpenProcess(
+                PROCESS_QUERY_INFROMATION, 0, pid
+            )
+            if processHandle == 0:
+                print('__running=0__')
+            else:
+                i = ctypes.c_int(0)
+                pi = ctypes.pointer(i)
+                if ctypes.windll.kernel32.GetExitCodeProcess(processHandle, pi) == 0:
+                    print('__running=0__')
+                if i.value == STILL_ACTIVE:
+                    print('__running=1__')
+                else:
+                    print('__running=0__')
+                ctypes.windll.kernel32.CloseHandle(processHandle)
+        else:
+            ps_cmd = f'ps -p {pid} > /dev/null && echo "__running=1__" || echo "__running=0__"'
```

Comment:
`os.kill(pid, 0)` should be a more reliable check for a process existing. Let's avoid os.system if we can.

Reply:
Interessting way of using the _kill_ command. Changed as suggested
https://github.com/ipython/ipyparallel/commit/ce43a8a784391e32659d94a54bd9ab748aafea76